src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -773,354 +764,353 @@ bool getDesuperHtrInput(EnergyPlusData &state)
+                DesupHtr.ValidSourceType = true;
+                HeatReclaim.ReclaimEfficiencyTotal += DesupHtr.HeatReclaimRecoveryEff;
                 if (HeatReclaim.ReclaimEfficiencyTotal > 0.3) {
                     ShowSevereError(state,
-                                    state.dataIPShortCut->cCurrentModuleObject + " = " +
-                                        state.dataWaterThermalTanks->WaterHeaterDesuperheater(DesuperheaterNum).Name +
+                                    state.dataIPShortCut->cCurrentModuleObject + " = " + DesupHtr.Name +
                                         ": "
                                         " sum of heat reclaim recovery efficiencies from the same source coil: \"" +
-                                        state.dataWaterThermalTanks->WaterHeaterDesuperheater(DesuperheaterNum).HeatingSourceName +
+                                        DesupHtr.HeatingSourceName + "\" cannot be over 0.3");
+                    ErrorsFound = true;
+                }
+            }
+        } else if (UtilityRoutines::SameString(heatSourceObjType, "Coil:Cooling:DX")) {
+            DesupHtr.ReclaimHeatingSource = CoilObjEnum::CoilCoolingDX;
+            DesupHtr.ReclaimHeatingSourceIndexNum = CoilCoolingDX::factory(state, state.dataIPShortCut->cAlphaArgs(10));
+            if (DesupHtr.ReclaimHeatingSourceIndexNum < 0) {
+                ShowSevereError(state,
+                                format("{}={}, could not find desuperheater coil {}={}",
+                                       state.dataIPShortCut->cCurrentModuleObject,
+                                       DesupHtr.Name,
+                                       state.dataIPShortCut->cAlphaArgs(9),
+                                       state.dataIPShortCut->cAlphaArgs(10)));
+                ErrorsFound = true;
+            } else {
+                DataHeatBalance::HeatReclaimDataBase &HeatReclaim =
+                    state.dataCoilCooingDX->coilCoolingDXs[DesupHtr.ReclaimHeatingSourceIndexNum].reclaimHeat;
+                if (!allocated(HeatReclaim.WaterHeatingDesuperheaterReclaimedHeat)) {
+                    HeatReclaim.WaterHeatingDesuperheaterReclaimedHeat.allocate(state.dataWaterThermalTanks->numWaterHeaterDesuperheater);
+                    for (auto &num : HeatReclaim.WaterHeatingDesuperheaterReclaimedHeat)
+                        num = 0.0;
+                }
+                DesupHtr.ValidSourceType = true;
+                HeatReclaim.ReclaimEfficiencyTotal += DesupHtr.HeatReclaimRecoveryEff;
+                if (HeatReclaim.ReclaimEfficiencyTotal > 0.3) {
+                    ShowSevereError(state,
+                                    state.dataIPShortCut->cCurrentModuleObject + ", \"" + DesupHtr.Name +
+                                        "\" sum of heat reclaim recovery efficiencies from the same source coil: \"" + DesupHtr.HeatingSourceName +
```

Comment:
Get input for new coil support in the water heater desuperheater.

Reply:
Also looks correct.